### ISSUE TITLE
bug report install-mount-rsync fails to find local ubuntu syslinux 6.03

### ISSUE THREAD

**Issue by dilbert-tctc** [jsamr/bootiso#21]
---------- bug report bootiso 3.1.1 and thinkpad-t470 loaded with ubuntu 18.04 

install-mount-rsync fails to find local ubuntu syslinux 6.03 

---------- check versions
<pre>
dilbert@thinkpad-t470:~$ bootiso -v
3.1.1
dilbert@thinkpad-t470:~$ uname -a
Linux thinkpad-t470 4.15.0-25-generic #27-Ubuntu SMP Mon Jul 2 08:35:25 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
dilbert@thinkpad-t470:~$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04 LTS
Release:	18.04
Codename:	bionic
</pre>
---------- install-auto works 
<pre>
dilbert@thinkpad-t470:~$ bootiso ubuntu-18.04-desktop-amd64.iso 
bootiso: Found hybrid ISO; choosing image copy `dd' mode.
bootiso: Partition label automatically set to `UBUNTU 18.0'.
         You can explicitly set label with `-L, --label' option.
bootiso: Listing drives available in your system:
         NAME   MODEL            VENDOR     SIZE TRAN   HOTPLUG
         sdc    Extreme          SanDisk   14.6G usb          1
bootiso: Autoselecting `sdc' (only USB device candidate)
bootiso: The selected device `/dev/sdc' is connected through USB.
bootiso: About to wipe out the content of device `/dev/sdc'.
         Are you sure you want to proceed? (y/n)>y
bootiso: Erasing contents of `/dev/sdc'...
bootiso: Copying files from ISO to USB device with `dd'    
bootiso: Synchronizing writes on device `/dev/sdc'    
bootiso: Took 56 seconds to perform [install-auto] action.
bootiso: USB device succesfully ejected.
         You can safely remove it!
</pre>
---------- install-mount-rsync fails to find local syslinux
<pre>
dilbert@thinkpad-t470:~$ bootiso --mrsync ubuntu-18.04-desktop-amd64.iso 
bootiso: UEFI boot check validated. Your USB will work with UEFI boot.
bootiso: Found SYSLINUX config file and binary at version 6.03.
bootiso: A SYSLINUX booloader will be installed on your USB.
bootiso: Partition label automatically set to `UBUNTU 18.0'.
         You can explicitly set label with `-L, --label' option.
bootiso: Listing drives available in your system:
         NAME   MODEL            VENDOR     SIZE TRAN   HOTPLUG
         sdc    Extreme          SanDisk   14.6G usb          1
bootiso: Autoselecting `sdc' (only USB device candidate)
bootiso: The selected device `/dev/sdc' is connected through USB.
bootiso: Created ISO mount point at `/mnt/iso.V1p'.
bootiso: About to wipe out the content of device `/dev/sdc'.
         Are you sure you want to proceed? (y/n)>y
bootiso: Erasing contents of `/dev/sdc'...
bootiso: Creating MBR partition table with `sfdisk' v2.31...
         Checking that no-one is using this disk right now ... OK
         
         Disk /dev/sdc: 14.6 GiB, 15693664256 bytes, 30651688 sectors
         Units: sectors of 1 * 512 = 512 bytes
         Sector size (logical/physical): 512 bytes / 512 bytes
         I/O size (minimum/optimal): 512 bytes / 512 bytes
         
         >>> Created a new DOS disklabel with disk identifier 0xdecf74ec.
         /dev/sdc1: Created a new partition 1 of type 'W95 FAT32 (LBA)' and of size 14.6 GiB.
         /dev/sdc2: Done.
         
         New situation:
         Disklabel type: dos
         Disk identifier: 0xdecf74ec
         
         Device     Boot Start      End  Sectors  Size Id Type
         /dev/sdc1  *     2048 30651687 30649640 14.6G  c W95 FAT32 (LBA)
         
         The partition table has been altered.
         Calling ioctl() to re-read partition table.
         Syncing disks.
bootiso: Creating vfat partition on `/dev/sdc1'...
         mkfs.fat 4.1 (2017-01-24)
         /dev/sdc1 has 64 heads and 32 sectors per track,
         hidden sectors 0x0800;
         logical sector size is 512,
         using 0xf8 media descriptor, with 30649640 sectors;
         drive number 0x80;
         filesystem has 2 32-bit FATs and 16 sectors per cluster.
         FAT size is 14960 sectors, and provides 1913730 clusters.
         There are 32 reserved sectors.
         Volume ID is b34f056a, volume label UBUNTU 18.0.
bootiso: Created USB device mount point at `/mnt/usb.Sgu'
bootiso: Copying files from ISO to USB device with `rsync'    
bootiso: Synchronizing writes on device `/dev/sdc'    
bootiso: Found ISOLINUX config file at `/mnt/usb.Sgu/isolinux/isolinux.cfg'.
         Moving to `/mnt/usb.Sgu/isolinux/syslinux.cfg'.
bootiso: Found local SYSLINUX version `6.03'
bootiso: ISO SYSLINUX version matches local version.
bootiso: Installing SYSLINUX bootloader in `/mnt/usb.Sgu/isolinux' with local version `6.03'...
         rsync: change_dir "/usr/lib/syslinux/bios" failed: No such file or directory (2)
         rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1196) [sender=3.1.2]
bootiso: SYSLINUX could not install C32 BIOS modules.
bootiso: C32 BIOS modules successfully installed.
         /usr/local/sbin/bootiso: line 1170: extlinux: command not found
bootiso: SYSLINUX bootloader could not be installed.
         Exiting...
bootiso: ISO succesfully unmounted (/mnt/iso.V1p).
bootiso: USB device partition succesfully unmounted.
</pre>
---------- install-mount-rsync works with remote-bootloader 6.03
<pre>
dilbert@thinkpad-t470:~$ bootiso --mrsync  --remote-bootloader 6.03 ubuntu-18.04-desktop-amd64.iso 
bootiso: UEFI boot check validated. Your USB will work with UEFI boot.
bootiso: Found SYSLINUX config file and binary at version 6.03.
bootiso: A SYSLINUX booloader will be installed on your USB.
bootiso: Partition label automatically set to `UBUNTU 18.0'.
         You can explicitly set label with `-L, --label' option.
bootiso: Listing drives available in your system:
         NAME   MODEL            VENDOR     SIZE TRAN   HOTPLUG
         sdc    Extreme          SanDisk   14.6G usb          1
bootiso: Autoselecting `sdc' (only USB device candidate)
bootiso: The selected device `/dev/sdc' is connected through USB.
bootiso: Created ISO mount point at `/mnt/iso.7XX'.
bootiso: About to wipe out the content of device `/dev/sdc'.
         Are you sure you want to proceed? (y/n)>y
bootiso: Erasing contents of `/dev/sdc'...
bootiso: Creating MBR partition table with `sfdisk' v2.31...
         Checking that no-one is using this disk right now ... OK
         
         Disk /dev/sdc: 14.6 GiB, 15693664256 bytes, 30651688 sectors
         Units: sectors of 1 * 512 = 512 bytes
         Sector size (logical/physical): 512 bytes / 512 bytes
         I/O size (minimum/optimal): 512 bytes / 512 bytes
         
         >>> Created a new DOS disklabel with disk identifier 0xd7909786.
         /dev/sdc1: Created a new partition 1 of type 'W95 FAT32 (LBA)' and of size 14.6 GiB.
         Partition #1 contains a vfat signature.
         The signature will be removed by a write command.
         /dev/sdc2: Done.
         
         New situation:
         Disklabel type: dos
         Disk identifier: 0xd7909786
         
         Device     Boot Start      End  Sectors  Size Id Type
         /dev/sdc1  *     2048 30651687 30649640 14.6G  c W95 FAT32 (LBA)
         
         Filesystem/RAID signature on partition 1 will be wiped.
         
         The partition table has been altered.
         Calling ioctl() to re-read partition table.
         Syncing disks.
bootiso: Creating vfat partition on `/dev/sdc1'...
         mkfs.fat 4.1 (2017-01-24)
         /dev/sdc1 has 64 heads and 32 sectors per track,
         hidden sectors 0x0800;
         logical sector size is 512,
         using 0xf8 media descriptor, with 30649640 sectors;
         drive number 0x80;
         filesystem has 2 32-bit FATs and 16 sectors per cluster.
         FAT size is 14960 sectors, and provides 1913730 clusters.
         There are 32 reserved sectors.
         Volume ID is 9bc1aea5, volume label UBUNTU 18.0.
bootiso: Created USB device mount point at `/mnt/usb.NeH'
bootiso: Copying files from ISO to USB device with `rsync'    
bootiso: Synchronizing writes on device `/dev/sdc'    
bootiso: Found ISOLINUX config file at `/mnt/usb.NeH/isolinux/isolinux.cfg'.
         Moving to `/mnt/usb.NeH/isolinux/syslinux.cfg'.
bootiso: Searching for SYSLINUX V6.03 remotely.
bootiso: Download of `/var/cache/bootiso/syslinux-6.03-pre20.tar.gz' completed (12M)
bootiso: SYSLINUX version `6.03' temporarily set for installation.
bootiso: Installing SYSLINUX bootloader in `/mnt/usb.NeH/isolinux' with kernel.org version `6.03'...
         /mnt/usb.NeH/isolinux is device /dev/sdc1
bootiso: Successfully installed SYSLINUX bootloader at version `6.03'.
bootiso: Succesfully installed Master Boot Record program.
bootiso: Took 64 seconds to perform [install-mount-rsync] action.
bootiso: ISO succesfully unmounted (/mnt/iso.7XX).
bootiso: USB device partition succesfully unmounted.
bootiso: USB device succesfully ejected.
         You can safely remove it!
</pre>
---------- install-mount-rsync works with remote-bootloader 6.04 
<pre>
dilbert@thinkpad-t470:~$ bootiso --mrsync  --remote-bootloader 6.04 ubuntu-18.04-desktop-amd64.iso 
[sudo] password for dilbert: 
bootiso: UEFI boot check validated. Your USB will work with UEFI boot.
bootiso: Found SYSLINUX config file and binary at version 6.03.
bootiso: A SYSLINUX booloader will be installed on your USB.
bootiso: Partition label automatically set to `UBUNTU 18.0'.
         You can explicitly set label with `-L, --label' option.
bootiso: Listing drives available in your system:
         NAME   MODEL            VENDOR     SIZE TRAN   HOTPLUG
         sdc    Cruzer           SanDisk    7.5G usb          1
bootiso: Autoselecting `sdc' (only USB device candidate)
bootiso: The selected device `/dev/sdc' is connected through USB.
bootiso: Created ISO mount point at `/mnt/iso.rrH'.
bootiso: About to wipe out the content of device `/dev/sdc'.
         Are you sure you want to proceed? (y/n)>y
bootiso: Erasing contents of `/dev/sdc'...
bootiso: Creating MBR partition table with `sfdisk' v2.31...
         Checking that no-one is using this disk right now ... OK
         
         Disk /dev/sdc: 7.5 GiB, 8036285952 bytes, 15695871 sectors
         Units: sectors of 1 * 512 = 512 bytes
         Sector size (logical/physical): 512 bytes / 512 bytes
         I/O size (minimum/optimal): 512 bytes / 512 bytes
         
         >>> Created a new DOS disklabel with disk identifier 0x490d80af.
         /dev/sdc1: Created a new partition 1 of type 'W95 FAT32 (LBA)' and of size 7.5 GiB.
         Partition #1 contains a vfat signature.
         The signature will be removed by a write command.
         /dev/sdc2: Done.
         
         New situation:
         Disklabel type: dos
         Disk identifier: 0x490d80af
         
         Device     Boot Start      End  Sectors  Size Id Type
         /dev/sdc1  *     2048 15695870 15693823  7.5G  c W95 FAT32 (LBA)
         
         Filesystem/RAID signature on partition 1 will be wiped.
         
         The partition table has been altered.
         Calling ioctl() to re-read partition table.
         Syncing disks.
bootiso: Creating vfat partition on `/dev/sdc1'...
         mkfs.fat 4.1 (2017-01-24)
         /dev/sdc1 has 248 heads and 62 sectors per track,
         hidden sectors 0x0800;
         logical sector size is 512,
         using 0xf8 media descriptor, with 15693823 sectors;
         drive number 0x80;
         filesystem has 2 32-bit FATs and 8 sectors per cluster.
         FAT size is 15304 sectors, and provides 1957897 clusters.
         There are 32 reserved sectors.
         Volume ID is aeb67e1b, volume label UBUNTU 18.0.
bootiso: Created USB device mount point at `/mnt/usb.Gg3'
bootiso: Copying files from ISO to USB device with `rsync'    
bootiso: Synchronizing writes on device `/dev/sdc'    
bootiso: Found ISOLINUX config file at `/mnt/usb.Gg3/isolinux/isolinux.cfg'.
         Moving to `/mnt/usb.Gg3/isolinux/syslinux.cfg'.
bootiso: Searching for SYSLINUX V6.04 remotely.
bootiso: Found `/var/cache/bootiso/syslinux-6.04-pre1.tar.gz' in cache.
bootiso: SYSLINUX version `6.04' temporarily set for installation.
bootiso: Installing SYSLINUX bootloader in `/mnt/usb.Gg3/isolinux' with kernel.org version `6.04'...
         /mnt/usb.Gg3/isolinux is device /dev/sdc1
bootiso: Successfully installed SYSLINUX bootloader at version `6.04'.
bootiso: Succesfully installed Master Boot Record program.
bootiso: Took 339 seconds to perform [install-mount-rsync] action.
bootiso: ISO succesfully unmounted (/mnt/iso.rrH).
bootiso: USB device partition succesfully unmounted.
bootiso: USB device succesfully ejected.
         You can safely remove it!
</pre>
---------- install-mount-rsync works with fedora 28 
<pre>
dilbert@thinkpad-t470:~$ bootiso --mrsync F28-WORK-x86_64-20180712-Live.iso 
[sudo] password for dilbert: 
bootiso: UEFI boot check validated. Your USB will work with UEFI boot.
bootiso: Found SYSLINUX config file and binary at version 6.04.
bootiso: A SYSLINUX booloader will be installed on your USB.
bootiso: Partition label automatically set to `F28-WORK-X8'.
         You can explicitly set label with `-L, --label' option.
bootiso: Listing drives available in your system:
         NAME   MODEL            VENDOR     SIZE TRAN   HOTPLUG
         sdc    DataTraveler 3.0 Kingston  28.9G usb          1
bootiso: Autoselecting `sdc' (only USB device candidate)
bootiso: The selected device `/dev/sdc' is connected through USB.
bootiso: Created ISO mount point at `/mnt/iso.iKV'.
bootiso: About to wipe out the content of device `/dev/sdc'.
         Are you sure you want to proceed? (y/n)>y
bootiso: Erasing contents of `/dev/sdc'...
bootiso: Creating MBR partition table with `sfdisk' v2.31...
         Checking that no-one is using this disk right now ... OK
         
         Disk /dev/sdc: 28.9 GiB, 31004295168 bytes, 60555264 sectors
         Units: sectors of 1 * 512 = 512 bytes
         Sector size (logical/physical): 512 bytes / 512 bytes
         I/O size (minimum/optimal): 512 bytes / 512 bytes
         
         >>> Created a new DOS disklabel with disk identifier 0x823839c1.
         /dev/sdc1: Created a new partition 1 of type 'W95 FAT32 (LBA)' and of size 28.9 GiB.
         Partition #1 contains a vfat signature.
         The signature will be removed by a write command.
         /dev/sdc2: Done.
         
         New situation:
         Disklabel type: dos
         Disk identifier: 0x823839c1
         
         Device     Boot Start      End  Sectors  Size Id Type
         /dev/sdc1  *     2048 60555263 60553216 28.9G  c W95 FAT32 (LBA)
         
         Filesystem/RAID signature on partition 1 will be wiped.
         
         The partition table has been altered.
         Calling ioctl() to re-read partition table.
         Syncing disks.
bootiso: Creating vfat partition on `/dev/sdc1'...
         mkfs.fat 4.1 (2017-01-24)
         /dev/sdc1 has 64 heads and 32 sectors per track,
         hidden sectors 0x0800;
         logical sector size is 512,
         using 0xf8 media descriptor, with 60553216 sectors;
         drive number 0x80;
         filesystem has 2 32-bit FATs and 32 sectors per cluster.
         FAT size is 14784 sectors, and provides 1891363 clusters.
         There are 32 reserved sectors.
         Volume ID is db802560, volume label F28-WORK-X8.
bootiso: Created USB device mount point at `/mnt/usb.QTJ'
bootiso: Copying files from ISO to USB device with `rsync'    
bootiso: Synchronizing writes on device `/dev/sdc'    
bootiso: Found ISOLINUX config file at `/mnt/usb.QTJ/isolinux/isolinux.cfg'.
         Moving to `/mnt/usb.QTJ/isolinux/syslinux.cfg'.
bootiso: Found local SYSLINUX version `6.03'
bootiso: ISO SYSLINUX version doesn't match local version.
         Scheduling download of version 6.04...
bootiso: Found `/var/cache/bootiso/syslinux-6.04-pre1.tar.gz' in cache.
bootiso: SYSLINUX version `6.04' temporarily set for installation.
bootiso: Installing SYSLINUX bootloader in `/mnt/usb.QTJ/isolinux' with kernel.org version `6.04'...
         /mnt/usb.QTJ/isolinux is device /dev/sdc1
bootiso: Successfully installed SYSLINUX bootloader at version `6.04'.
bootiso: Succesfully installed Master Boot Record program.
bootiso: Took 142 seconds to perform [install-mount-rsync] action.
bootiso: ISO succesfully unmounted (/mnt/iso.iKV).
bootiso: USB device partition succesfully unmounted.
bootiso: USB device succesfully ejected.
         You can safely remove it!
</pre>
---------- the end 

we have needed a great cli tool like this for a long time! thanks for sharing!

**Comment by jsamr** [comment]
OK, thanks for the report and the kudos!
 `extlinux: command not found` is surprising, since `extlinux` should be shipped with `syslinux`.  
I will add a check here anyway.
Can you confirm you're running Ubuntu 18.04 ?
If so, could you provide the output of `apt-file list syslinux` to see what files package `syslinux` owns?

**Comment by dilbert-tctc** [comment]
dilbert@thinkpad-t470:~$ apt-file list syslinux
syslinux: /usr/bin/syslinux
syslinux: /usr/lib/SYSLINUX/altmbr.bin
syslinux: /usr/lib/SYSLINUX/gptmbr.bin
syslinux: /usr/lib/SYSLINUX/mbr.bin
syslinux: /usr/share/doc/syslinux/README.txt.gz
syslinux: /usr/share/doc/syslinux/changelog.Debian.gz
syslinux: /usr/share/doc/syslinux/copyright
syslinux: /usr/share/lintian/overrides/syslinux
syslinux: /usr/share/man/man1/syslinux.1.gz

**Comment by dilbert-tctc** [comment]
dilbert@thinkpad-t470:~$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04 LTS
Release:	18.04
Codename:	bionic

**Comment by dilbert-tctc** [comment]
dilbert@thinkpad-t470:~$ uname -a
Linux thinkpad-t470 4.15.0-25-generic #27-Ubuntu SMP Mon Jul 2 08:35:25 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

**Comment by dilbert-tctc** [comment]
possible error is
rsync: change_dir "/usr/lib/syslinux/bios" failed: No such file or directory 
as my local tree is
usr/lib/syslinux/modules/bios/...

**Comment by jsamr** [comment]
Thanks. I'll dig this and handle more carefully distro variations around syslinux deployment :-)

**Comment by dilbert-tctc** [comment]
thank you!

**Comment by HassanMirza01** [comment]
Infected from same error, Exactly same, i modified code myself but still there is no bios folder where its looking for, install-auto, install-dd both works fine.... i think we have to use the windows's isos method on linux iso's too, i will try to add gpt support in it too, 

cool cli tool, thanks though ;)

**Comment by jsamr** [comment]
@HassanMirza01 Sorry about that ; I'll check that tonight!

**Comment by HassanMirza01** [comment]
not an issue, i figured it out myself, actually mbr.bin file is in lib/syslinux/mbr/ not in bios/mbr/
and also, due to this path issue, dd command also failed while installing mbr in usb, new bios folder is in syslinux/modules/ and not in syslinux/ ;)

**Comment by HassanMirza01** [comment]
@jsamr even though i didn't understood that extbin thing but my iso of debian booting fine with correct syslinux install with install-mount-rsync option, still in burning iso, i can see errors of extbin but c32 bios modules installed successfully... i think we dont need extbin at all, i didn't find anything regarding it on web even....

<pre>
bootiso: Created USB device mount point at '/mnt/usb.jtI'

bootiso: Copying files from ISO to USB device with 'rsync'    

bootiso: Synchronizing writes on device '/dev/sdb'   -
 
bootiso: Found local SYSLINUX version '6.03'
bootiso: ISO SYSLINUX version matches local version.
bootiso: Installing SYSLINUX bootloader in '/mnt/usb.jtI/arch/boot/syslinux' with local version '6.03'...
bootiso: C32 BIOS modules successfully installed.
/home/hassanmirza01/bin/bootiso: line 677: 'extBin': syntax error: operand expected (error token is "'extBin'")
/home/hassanmirza01/bin/bootiso: line 938: wait_for: No record of process 4342
         bootiso: ISO succesfully unmounted (/mnt/iso.6WB).
         bootiso: USB device partition succesfully unmounted.
bootiso: Successfully installed SYSLINUX bootloader at version '6.03'.
         dd: failed to open 'extlinux': No such file or directory
bootiso: Succesfully installed Master Boot Record program.
bootiso: Took 120 seconds to perform Required action.
         
bootiso: USB device successfully ejected.
         You can safely remove it!
         
hassanmirza01@bionic:~$
</pre>

**Comment by jsamr** [comment]
@HassanMirza01 `extBin` is a local variable containing the file location of `extlinux` binary, which should be shipped with any `syslinux` package but apparently it is not the case on ubuntu systems. [See official wiki :-)](https://www.syslinux.org/wiki/index.php?title=EXTLINUX). Getting my hands dirty and fixing this issue this morning.

**Comment by HassanMirza01** [comment]
Well, not an issue, for me, linux iso's working fine with both methods, even though extbin errors came but i will remove their lines etc, 

Am focussing on windows iso stuff now, just noticed that ms-sys discontinued long ago, find a way to get it done ;)

**Comment by jsamr** [comment]
@HassanMirza01 That is probably because the ISO you're referring to supports UEFI as your BIOS do.  Syslinux job is to install a legacy BIOS bootloader (and the MBR program) to boot on older devices. You can confirm this with `bootiso --inspect my-iso-file.iso` :-) Anyway, I'm glad it works out for you in that peculiar case but I must fix it anyway ^^

**Comment by HassanMirza01** [comment]
my laptop support both legacy and uefi boots, but windows iso's need to support correct one according to partition scheme of hdd, in that case, its GPT, so i need uefi one, mbr is old fashioned now, i have found a way to achieve gpt stuff too... i will may be made a pull request for bootiso ;)

**Comment by jsamr** [comment]
@HassanMirza01 OK, see if your experience with Windows is related to #15. For any pull requests, I suggest we discuss first some details in [the gitter channel](https://gitter.im/bootiso) I've just open on purpose!

**Comment by HassanMirza01** [comment]
i succeeded in booting GPT burned on my Ubuntu 18.04, and its showing drives and i know that issue, drivers missing it says if we boot mbr on gpt or gpt on mbr etc, but i succeeded in achieving it... tried windows 10 1607 LTSB and it worked perfectly, adding in my program, actually i wrote my own script named BurnISO but i was missing some commands which i gets from your script, commands like syslinux related, i have added gpt thing in that and its working like a charm ;)

**Comment by jsamr** [comment]
@dilbert-tctc @HassanMirza01 
I've committed a fix.
Could you try with the next release candidate to see if the missing file errors have disappeared ?

``` bash
curl -L https://rawgit.com/jsamr/bootiso/v3.1.2-rc-1/bootiso -O
chmod +x bootiso
```

**Comment by HassanMirza01** [comment]
man, you are doing it more complicated... installing mbr is just one command thing, thats why i removed all syslinux stuff i added from your script, its not a big thing though....

**Comment by jsamr** [comment]
@HassanMirza01 `SYSLINUX` is necessary for legacy boot, and is shipped in many distro ISOs.
I'm not sure what are you referring to when you mention "complicated", but there might be several aspects that can participate in that complexity:
- when local version of SYSLINUX doesn't match the ISO version, bootiso downloads the right version locally from kernel.org. This is because I have witnessed unbooting USB sticks in such situations. Such examples include openSUSE-Tumbleweed, shipped with v4.04 and elementaryOS, shipped with v4.05 (current stable is 6.03).
- SYSLINUX packaging differs from one distro to another. The "complexity" I've added in this commit with `checkSyslinuxInstall` is to find the appropriate files.

Perhaps you have now more elements in mind to understand the need for such "complexity", but if you don't, and if you find there is a better, simpler way to achieve the same results, with the same match of compatible systems, I'm very open to hear it.

**Comment by HassanMirza01** [comment]
Cool bro... I understood it well now, actually your install mbr program is too long may be thats why i responded like that, also, i never used other distros than *ubuntu,

**Comment by jsamr** [comment]
@HassanMirza01 np :-)

**Comment by jsamr** [comment]
@HassanMirza01 It would be very nice of you to check there are no more errors with the v3.1.2-rc-1!

**Comment by jsamr** [comment]
I've tested in an Ubuntu VM and the issue did not occur. Closed. Fixed in latest release `v3.1.3`

**Comment by dilbert-tctc** [comment]
I just did a quick check with Ubuntu and it worked for me. Thanks again!
